RFS/src/main/java/org/opensearch/migrations/cluster/ClusterProviderRegistry.java

```diff
@@ -39,15 +40,21 @@ private List<VersionSpecificCluster> getProviders() {
      * @param repo The source repo that contains of the snapshot
      * @return The snapshot resource provider
      */
-    public ClusterSnapshotReader getSnapshotReader(Version version, SourceRepo repo) {
+    public ClusterSnapshotReader getSnapshotReader(Version version, SourceRepo repo, boolean looseMatch) {
```

Comment:
nit: seems you could have this as an overloaded method, where looseMatch is false by default to avoid your touch points here